source/features/ci-link.tsx

```diff
@@ -54,22 +56,74 @@ async function init(signal: AbortSignal): Promise<void> {
 		// Old selector: `.avatar` excludes "Global navigation update"
 		// Repo title (aware of forks and private repos)
 		'[itemprop="name"]:not(.avatar ~ [itemprop])',
-	], add, {signal});
+	], addRepoIcon, {signal});
+}
+
+function addPRIcon(base?: DocumentFragment | HTMLElement): void {
+	const selectBase = base instanceof DocumentFragment ? base : document.documentElement;
+	const iconWrapper = lastElement('.TimelineItem .js-socket-channel[data-url]', selectBase);
+	if (!iconWrapper) {
+		return;
+	}
+
+	$('#partial-discussion-header')!.classList.add('rgh-pr-ci-link-added');
+
+	const ciLinkCommitSha = /[a-f\d]{40}/.exec(iconWrapper.dataset.url!)![0];
```

Comment:
Use the commit OID to check if the CI link needs updating. Sometimes it's an URL parameter, sometimes it's in the pathname.